lib/lightspeed/client.rb

```diff
@@ -26,25 +29,38 @@ def load_json(json)
     end
 
     def get(**args)
-      request(args.merge(method: :get)).perform
+      perform_request(:get, 1, **args)
     end
 
     def post(**args)
-      request(args.merge(method: :post)).perform
+      perform_request(:post, 10, **args)
     end
 
     def put(**args)
-      request(args.merge(method: :put)).perform
+      perform_request(:put, 10, **args)
     end
 
     def delete(**args)
-      request(args.merge(method: :delete)).perform
+      perform_request(:delete, 10, **args)
     end
 
     private
 
+    def perform_request(method, units, **args)
```

Comment:
I'm not sure I follow what you mean. Semantically, the API limits (bucket level and max) belong to an account as a whole and not just one request. If I tell the request to do the thing and it decides not to, some mechanism needs to keep further requests from trying to do their things, too.

A request by itself can't decide if its about to overflow the bucket, since it doesn't know how full it is until it tries.